Sources/Core/AWSClientRuntime/EventStream/AWSMessageDecoderStream.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+
+extension AWSEventStream {
+
+    /// Stream adapter that decodes input data into `EventStream.Message` objects.
+    struct AWSMessageDecoderStream<Element: MessageUnmarshallable>: MessageDecoderStream {
```

Comment:
Would `Event` be a better name here? 

```suggestion
    struct AWSMessageDecoderStream<Event: MessageUnmarshallable>: MessageDecoderStream {
```

Reply:
that's fair, when you use Element, you don't have to mapping between the AsyncSequence and other types but that's fine.